### DISCUSSION TITLE
Need help with a chatbot

### DISCUSSION THREAD

**Discussion by dramirz** [beeware/toga#2794] [Debugging help]
My project is a chatbot.
my chatbot works fine.
user can input text and get a response.
however, there's one thing I can't figure out. I want the scroll bar to go down automatically after hitting the send button. i want the text output to be visible without having to scroll. normally, when hitting send, the output is not visible. user needs to scroll down to see it.

I have tried the following code under the method that handles the send button.
self.scroll_to_bottom()
    def scroll_to_bottom(self):  
        # Scroll to the bottom by setting the vertical scroll position to 1
        self.scroll_container.vertical_position = 1 

this is my code for the scroll container:
 # Create the ScrollContainer for the chat output
        self.chat_output = MultilineTextInput(readonly=True, style=Pack(flex=1, padding=10))
        self.scroll_container = ScrollContainer(content=self.chat_output, style=Pack(flex=1))
        main_box.add(self.scroll_container)
![Capture](https://github.com/user-attachments/assets/978f34c9-77f4-4b6a-a9e9-054e08a75544)

If you have any questions about my code or need more information, I will be happy to provide it. Thanks.

**Comment by freakboy3742** [comment]
I think you're getting stuck on having multiple scroll bars in play.

Based on your example code, you've put the MultilineTextInput inside a scroll container. This shouldn't be necessary - a MultilineTextInput will have a scroll bar if there's too much content to fit. In your layout, you're putting that scrollable content *inside* a scroll view - so it's not clear *which* scroll bar you're moving when you add content.

MultilineTextInput also has `scroll_to_bottom()` method to push the scrollable area so the bottom text is visible.

So - you should be able to add the MultilineTextInput directly to the main box, and call `self.chat_output.scroll_to_bottom()` after adding a new chat sequence.

As an aside, you may also want to look at the `Beeliza` example in the Toga examples folder - that's an implementation of a chatbot, using the `DetailedList` widget. That won't work quite as well on Windows due to how DetailedList is currently implemented on Windows; but on other platforms, it will give you a much nicer user experience.
